GVFS/GVFS.Service/ProductUpgradeTimer.cs

```diff
@@ -20,15 +20,21 @@ public ProductUpgradeTimer(JsonTracer tracer)
 
         public void Start()
         {
-            Random random = new Random();
-            TimeSpan startTime = TimeSpan.Zero;
+            if (!GVFSEnlistment.IsUnattended(this.tracer))
+            {
+                TimeSpan startTime = TimeSpan.Zero;
 
-            this.tracer.RelatedInfo("Starting auto upgrade checks.");
-            this.timer = new Timer(
-                this.TimerCallback,
-                state: null,
-                dueTime: startTime,
-                period: TimeInterval);
+                this.tracer.RelatedInfo("Starting auto upgrade checks.");
+                this.timer = new Timer(

```

Comment:
Unrelated to this change, I noticed that ProductUpgradeTimer doesn't implement `IDisposable`. Should it?

Reply:
Makes sense. Added Disposable.